Desktop/widgets/factorsformbase.cpp

```diff
@@ -133,19 +132,16 @@ void FactorsFormBase::termsChangedHandler()
 	_boundTo->setValue(allOptions);	
 }
 
-void FactorsFormBase::addListViewSlot(JASPListControl *listView)
+void FactorsFormBase::factorsAddedSlot(int index, QVariant item)
 {
-	const vector<tuple<string, string, vector<string> > > &factors = _factorsModel->getFactors();
-	QStringList names;
-	for (auto factor : factors)
-		names.push_back(tq(get<0>(factor)));
-	_availableVariablesListItem->setProperty("dropKeys", names);
-	setProperty("dropKeys", _availableVariablesListName);
+	VariablesListBase* listView = qobject_cast<VariablesListBase *>(item.value<QObject *>());
+	if (!listView)
+	{
+		Log::log() << "JASP Control is not a VariablesListBase in factorAdded" << std::endl;
```

Comment:
Maybe also mention the `name` of the control in this (and these kind of) message(s). That way the log is slightly *more* useful.